docs/af5-app-from-scratch/modules/ROOT/pages/implement-command-subscribe-student.adoc

```diff
@@ -0,0 +1,601 @@
+<2> Thanks to the `SubscriptionId` which is composed of `courseId` and `studentId` we know the values of those tags we needed.
+<3> As you may see at the highest level those `EventCriteria` are combined with `either` method, but when we define tags in `withTags` it means that the certain type of event requires all of them (there is an OR relation between event types, an AND relation between tags and OR between criteria).
+If we do `.withTags(Tag.of("courseId", courseId), Tag.of("studentId", studentId))` we will receive just subscription event of the given student for just one given course.
+It's not what we wanted here.
+So, we split `StudentSubscribed` and `StudentUnsubscribed` events into two separate criteria (one for student and one for course), because we need to load all events of those types for either `courseId` or `studentId`.
+
+It gives us better accessibility of our system - thanks to that, as you see there is no `CourseRenamed` event in our criteria, so the Faculty administrator is still able to rename the course in the same time while processing the `SubscribeStudent` command, because the `CourseRenamed` event doesn't match the criteria - it's not in our operation consistency boundary.
+In case of Aggregates, those operations may clash, or you need to introduce separated entity for the name if you want to avoid concurrency access issues.
+Our colleague Milan from AxonIQ (with our ex-colleague Sara) discuss those scenarios in the talk—we really encourage you to watch it link:https://www.youtube.com/watch?v=IgigmuHHchI[The Aggregate is dead. Long live the Aggregate! by Sara Pellegrini & Milan Savic @ Spring I/O 2023].
+
+[NOTE]
+====
+Keep in mind it's beneficial to define events types in the criteria.
+Technically you can just use `EventCriteria.match().eventsOfTypes(...)` and load all events for the given tag.
+But in this case, you load more than needed, and it's possible to clash with other changes, which cause events that do not influence these certain business rules.
+====
+
+== Summing up
+
+
+Let's summarize what we have done so far.
+We've implemented the whole `SubscribeStudent` command handler using the DCB concept in practice.
+It was easier than you expected, right?
+If you're not sure if you followed the tutorial correctly, you can always check the code in the repository. The command handler code is here link:https://github.com/AxonIQ/university-demo/blob/master/src/main/java/io/axoniq/demo/university/faculty/write/subscribestudent/SubscribeStudentCommandHandler.java[SubscribeStudentCommandHandler].
+If you prefer to use different style (with multiple state classes - like `Course` and `Student` instead of just one) you may also compare the solution we have done with the code in the package link:https://github.com/AxonIQ/university-demo/tree/master/src/main/java/io/axoniq/demo/university/faculty/write/subscribestudentmulti[subscribestudentmulti].
```

Comment:
```suggestion
If you prefer to use a different style (with multiple state classes - like `Course` and `Student` instead of just one) you may also compare it with the solution we have done in the link:https://github.com/AxonIQ/university-demo/tree/master/src/main/java/io/axoniq/demo/university/faculty/write/subscribestudentmulti[subscribestudentmulti] package.
```